test/e2e/storage/testsuites/volumelimits.go

```diff
@@ -0,0 +1,335 @@
+						}
+					}
+				}
+				if existing > 0 {
+					e2elog.Logf("Waiting for %d PVs to be deleted", existing)
+					return false, nil
+				}
+				return true, nil
+			})
+			if err != nil {
+				cleanupErrors = append(cleanupErrors, fmt.Sprintf("timed out waiting for PVs to be deleted: %s", err))
+			}
+			if len(cleanupErrors) != 0 {
+				framework.ExpectNoError(fmt.Errorf("test cleanup failed: " + strings.Join(cleanupErrors, "; ")))
+			}
+		}()
+
+		// Create <limit> PVCs for one gigantic pod.
+		for i := 0; i < limit; i++ {
+			ginkgo.By(fmt.Sprintf("Creating volume %d/%d", i+1, limit))
+			pvc := framework.MakePersistentVolumeClaim(framework.PersistentVolumeClaimConfig{
+				ClaimSize:        dDriver.GetClaimSize(),
+				StorageClassName: &l.resource.sc.Name,
+			}, l.ns.Name)
+			pvc, err = l.cs.CoreV1().PersistentVolumeClaims(l.ns.Name).Create(pvc)
+			framework.ExpectNoError(err)
+			l.pvcs = append(l.pvcs, pvc)
+		}
+
+		// Create the gigantic pod with all <limit> PVCs
+		ginkgo.By("Creating pod")
+		pod := framework.MakeSecPod(l.ns.Name, l.pvcs, nil, false, "", false, false, framework.SELinuxLabel, nil)
+		// Use affinity to schedule everything on the right node
+		selection := framework.NodeSelection{}
+		framework.SetAffinity(&selection, nodeName)
+		pod.Spec.Affinity = selection.Affinity
+		l.runningPod, err = l.cs.CoreV1().Pods(l.ns.Name).Create(pod)
+		framework.ExpectNoError(err)
+
+		l.pvNames = sets.NewString()
+		ginkgo.By("Waiting for all PVCs to get Bound")
+		err = wait.Poll(5*time.Second, testSlowMultiplier*framework.PVBindingTimeout, func() (bool, error) {
+			unbound := 0
+			for _, pvc := range l.pvcs {
+				pvc, err := l.cs.CoreV1().PersistentVolumeClaims(l.ns.Name).Get(pvc.Name, metav1.GetOptions{})
+				if err != nil {
+					return false, err
+				}
+				if pvc.Status.Phase != v1.ClaimBound {
+					unbound++
+				} else {
+					l.pvNames.Insert(pvc.Spec.VolumeName)
+				}
+			}
+			if unbound > 0 {
+				e2elog.Logf("%d/%d of PVCs are Bound", l.pvNames.Len(), limit)
+				return false, nil
+			}
+			return true, nil
+		})
+		framework.ExpectNoError(err)
+
+		e2elog.Logf("All volumes are bound")
+		ginkgo.By("Waiting for the pod Running")
+		err = e2epod.WaitTimeoutForPodRunningInNamespace(l.cs, l.runningPod.Name, l.ns.Name, testSlowMultiplier*framework.PodStartTimeout)
+		framework.ExpectNoError(err)
+
+		ginkgo.By("Creating an extra pod with one volume to exceed the limit")
+		pod = framework.MakeSecPod(l.ns.Name, []*v1.PersistentVolumeClaim{l.resource.pvc}, nil, false, "", false, false, framework.SELinuxLabel, nil)
+		// Use affinity to schedule everything on the right node
+		framework.SetAffinity(&selection, nodeName)
+		pod.Spec.Affinity = selection.Affinity
+		l.unschedulablePod, err = l.cs.CoreV1().Pods(l.ns.Name).Create(pod)
+
+		ginkgo.By("Waiting for the pod to get unschedulable")
+		err = wait.Poll(5*time.Second, framework.PodStartTimeout, func() (bool, error) {
```

Comment:
helper function?

Reply:
Found `WaitForPodCondition`, but it's not that shorter.